src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,37 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int INDEX_ZERO = 0;
+    private static final int INDEX_ONE = 1;
+    private static final int INDEX_TWO = 2;
+    private static final int INDEX_THREE = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+        int[] sum = new int[names.length];
+        LocalDate beginningOfPeriod = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate endOfPeriod = LocalDate.parse(dateTo,FORMATTER);
+        for (int i = 0; i < data.length; i++) {
+            String[] dataArray = data[i].split(" ");
+            LocalDate workingDay = LocalDate.parse(dataArray[INDEX_ZERO], FORMATTER);
+            if (!workingDay.isBefore(beginningOfPeriod) && !workingDay.isAfter(endOfPeriod)) {
+                for (int k = 0; k < names.length; k++) {
```

Comment:
If you change your logic to iterate by **names** first and then by **data** you can rid of 3rd loop